### PR TITLE
Added required sections in sympy.combinatorics 

### PR DESCRIPTION
**Pull request by Soumi7**
# Updates sympy.combinatorics to follow latest docstring style guide

#### References to other Issues or PRs

#19591

#### Brief description of what is fixed or changed
- Required sections in :
  - pc_groups.py
  - permutataions.py
  - perm_groups.py

#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/combinatorics/perm_groups.py [hunk 1]
```diff
@@ -2444,8 +2525,8 @@ def lower_central_series(self):
     def max_div(self):
         """Maximum proper divisor of the degree of a permutation group.
 
-        Notes
-        =====
+        Explanations
+        ============
```
Comment (oscarbenjamin) [inline sympy/combinatorics/perm_groups.py:2529]: Explanation

File: sympy/combinatorics/perm_groups.py [hunk 2]
```diff
@@ -1449,14 +1506,17 @@ def derived_subgroup(self):
...
+        Explanation
+        ===========
 
         Iteration is done with one of these methods::
 
           method='coset'  using the Schreier-Sims coset representation
           method='dimino' using the Dimino method
 
-        If af = True it yields the array form of the permutations
+        If ``af`` = ``True`` it yields the array form of the permutations
```
Comment (sylee957) [inline sympy/combinatorics/perm_groups.py:1519]:
```suggestion
        If ``af = True`` it yields the array form of the permutations
```

File: sympy/combinatorics/perm_groups.py (hunk 1 above)
Comment (Soumi7) [reply sympy/combinatorics/perm_groups.py]: Corrected.

File: sympy/combinatorics/perm_groups.py (hunk 2 above)
Comment (Soumi7) [reply sympy/combinatorics/perm_groups.py]: Changed.

File: sympy/combinatorics/perm_groups.py (hunk 1 above)
Comment (oscarbenjamin) [reply sympy/combinatorics/perm_groups.py]: There are more examples of this.

**Comment by Sc0rpi0n101** [comment]
Looks good to me. If anyone else doesn't have any comments, can you squash your commits?

**Comment by Soumi7** [comment]
I had been working on multiple PRs at the same time as the old ones werent receiving comments immediately. So the previous commits were on multiple branches, squashing the commits would be an issue. Could you merge the latest two PRs? Then I will make a fresh PR.

**Comment by Sc0rpi0n101** [comment]
Sure, merging this now then 🎉 
Thank you for your contribution.

**Pull request merged**